File: android/app/src/main/java/app/gsuss/asistente/BankApps.kt
package app.gsuss.asistente

import java.text.Normalizer

/** Blocks bank / bank-wallet apps. Launch still uses the system UI. */
object BankApps {
    private val shortTokens = listOf(
        "bank", "bna", "bbva", "hsbc", "icbc", "itau", "n26", "modo", "uala",
        "reba", "prex", "wise", "bind",
    )
    private val longNeedles = listOf(
        "banco", "bancar", "banking", "mbanking", "mercadopago", "mercado pago",
        "brubank", "naranjax", "naranja x", "galicia", "santander", "hipotecario",
        "credicoop", "supervielle", "personalpay", "personal pay", "rebanking",
        "paypal", "revolut", "citibank", "openbank", "nubank", "cuentadni",
        "bnaplus", "banca movil", "bancamovil", "homebanking", "home banking",
        "walletnfcrel", "comafi", "patagoniabank",
    )

    fun fold(text: String): String {
        val nfd = Normalizer.normalize(text.lowercase(), Normalizer.Form.NFD)
        return nfd.replace("\\p{Mn}+".toRegex(), "")
    }

    fun blocked(vararg parts: String): Boolean {
        val blob = fold(parts.joinToString(" "))
        if (blob.isBlank()) return false
        val packed = blob.replace(" ", "")
        for (needle in longNeedles) {
            val n = fold(needle)
            if (n in blob || n.replace(" ", "") in packed) return true
        }
        for (token in shortTokens) {
            if (Regex("(^|[^a-z0-9])${Regex.escape(token)}([^a-z0-9]|$)").containsMatchIn(blob)) {
                return true
            }
        }
        return false
    }
}

File: jarvis/bank_apps.py
"""Refuse banking / payment-bank apps. Names only; no bypass of OS auth."""

from __future__ import annotations

import re
import unicodedata

_SHORT = (
    "bank",
    "bna",
    "bbva",
    "hsbc",
    "icbc",
    "itau",
    "n26",
    "modo",
    "uala",
    "reba",
    "prex",
    "wise",
    "bind",
)

_LONG = (
    "banco",
    "bancar",
    "banking",
    "mbanking",
    "mercadopago",
    "mercado pago",
    "brubank",
    "naranjax",
    "naranja x",
    "galicia",
    "santander",
    "hipotecario",
    "credicoop",
    "supervielle",
    "personalpay",
    "personal pay",
    "rebanking",
    "paypal",
    "revolut",
    "citibank",
    "openbank",
    "nubank",
    "cuentadni",
    "bnaplus",
    "banca movil",
    "bancamovil",
    "homebanking",
    "home banking",
)

_PACK_BITS = (
    "mercadopago",
    "brubank",
    "naranjax",
    "uala",
    "galicia",
    "santander",
    "bbva",
    "hsbc",
    "icbc",
    "bancamovil",
    "mbanking",
    "personalpay",
    "rebanking",
    "paypal",
    "revolut",
    "walletnfcrel",
    "bnaplus",
    "hipotecario",
    "credicoop",
    "supervielle",
    "openbank",
    "nubank",
    "cuentadni",
    "comafi",
    "patagoniabank",
    "banco",
    ".bank.",
    "banking",
)


def fold(text: str) -> str:
    raw = unicodedata.normalize("NFD", (text or "").lower())
    return "".join(ch for ch in raw if unicodedata.category(ch) != "Mn")


def is_banking(name: str) -> bool:
    blob = fold(name)
    if not blob.strip():
        return False
    packed = blob.replace(" ", "")
    for bit in _PACK_BITS:
        if bit in blob or bit in packed:
            return True
    for needle in _LONG:
        if needle in blob or needle.replace(" ", "") in packed:
            return True
    for token in _SHORT:
        if re.search(rf"(^|[^a-z0-9]){re.escape(token)}([^a-z0-9]|$)", blob):
            return True
    return False
